Feat: Add google_oracle_database_db_nodes datasource (#11988)
GoogleCloudPlatform/magic-modules: mmv1/third_party/terraform/website/docs/d/oracle_database_db_nodes.html.markdown

---
subcategory: "Oracle Database"
description: |-
  List all database nodes of a Cloud VmCluster.
---

# google_oracle_database_db_nodes

> List all DbNodes of a Cloud VmCluster.

Reviewer: Please add a link to the cloud.google.com REST API documentation for this endpoint.

Author: This is read only resource, added API link. Thank you for the suggestions.

## Example Usage

```hcl
data "google_oracle_database_db_nodes" "my_db_nodes"{
	location = "us-east4"
	cloud_vm_cluster = "vmcluster-id"
}
```

## Argument Reference

The following arguments are supported:

* `cloud_vm_cluster` - (Required) The ID of the VM Cluster.

* `location` - (Required) The location of the resource.

- - -
* `project` - (Optional) The project in which the resource belongs. If it
    is not provided, the provider project is used.

> * `db_nodes` - (Output) List of dbNodes.

Reviewer: Missing "structure is documented below" link - see https://github.com/hashicorp/terraform-provider-google/blob/e65430eedacbbd7a89d31208ef578840b53054e7/website/docs/d/dns_keys.html.markdown?plain=1#L55 for example.

Note that the nested properties should come after all properties at a given level to avoid ambiguity about whether something is nested or not -  in this case it looks like the order is unambiguous anyway but it would be nice to have the links for consistency.

<a name="nested_properties"></a> The `db_nodes` block supports:

* `name` - User friendly name for the resource.

> * `properties` - Various properties of the databse node.

Reviewer: Should also have a link for consistency.

> * `properties` - Various properties of the databse node.

Reviewer: ```suggestion
* `properties` - Various properties of the database node.
```

---
subcategory: "Oracle Database"
description: |-
  List all database nodes of a Cloud VmCluster.
---

# google_oracle_database_db_nodes

List all DbNodes of a Cloud VmCluster.

For more information see the
[API](https://cloud.google.com/oracle/database/docs/reference/rest/v1/projects.locations.cloudVmClusters.dbNodes).

## Example Usage

```hcl
data "google_oracle_database_db_nodes" "my_db_nodes"{
	location = "us-east4"
	cloud_vm_cluster = "vmcluster-id"
}
```

## Argument Reference

The following arguments are supported:

* `cloud_vm_cluster` - (Required) The ID of the VM Cluster.

* `location` - (Required) The location of the resource.

> - - -

Reviewer: ```suggestion

```

* `project` - (Optional) The project in which the resource belongs. If it
    is not provided, the provider project is used.

> * `db_nodes` - (Output) List of dbNodes. Structure is [documented below](#nested_dbnodes).

Reviewer: this is closer to "standard" for datasources. For example, see https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/bigquery_table_iam_policy

```suggestion
## Attributes reference

The following attributes are exported:

* `db_nodes` - List of dbNodes. Structure is [documented below](#nested_dbnodes).
```

<a name="nested_dbnodes"></a> The `db_nodes` block supports:

> * `name` - User friendly name for the resource.

Reviewer: According to the API docs this should be:

```suggestion
* `name` - The name of the database node resource in the following format: projects/{project}/locations/{location}/cloudVmClusters/{cloudVmCluster}/dbNodes/{db_node}
```

* `properties` - Various properties of the database node. Structure is [documented below](#nested_properties).

<a name="nested_properties"></a> The `properties` block supports:

* `ocid`- OCID of database node.

* `ocpu_count` - OCPU count per database node.

* `memory_size_gb` - The allocated memory in GBs on the database node.

* `db_node_storage_size_gb` - The allocated local node storage in GBs on the database node.

* `db_server_ocid` - The OCID of the Database server associated with the database node.

* `hostname` - The host name for the database node.

* `state` - State of the database node.

> <a name="nested_states"></a>Allowed values for `state` are:<br>

Reviewer: Nit: "Possible" might be better since they're output values (so from the user perspective "allowed" is not meaningful.)

```suggestion
<a name="nested_states"></a>Possible values for `state` are:<br>
```

> `STATE_UNSPECIFIED` - Default unspecified value.<br>

Reviewer: Generally we don't document the unspecified state since it's not useful.

```suggestion
```
